lib/realtime/tenants/check.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Realtime.Tenants.Check do
+  use GenServer
+
+  require Logger
+
+  alias Realtime.Helpers
+
+  def connection_status(tenant_id) do
+    case get_status(tenant_id) do
+      :undefined ->
+        :ok
+        node = Realtime.Nodes.get_node_for_tenant_id(tenant_id)
+
+        case :rpc.call(node, __MODULE__, :set_status, [tenant_id]) do
```

Comment:
So here the logic is?:
- `get_status` from syn
- if we don't have it yet - RPC to set set it via syn
- then `get_status` from syn again

If we don't have it already and you immediately look for it again you won't have it either because it's going to take some milliseconds to propagate to the caller node.

Reply:
my plan would be to retry with a backoff (logic is underneath) until we have the information in :syn since we're working with eventually consistency 😓

Reply:
Yeah that should work. Didn't fully look at how your backoff stuff works yet honestly.